### PR TITLE
added MERGE object function

### PR DESCRIPTION
**Pull request by 3timeslazy**

### CODE REVIEW

**Comment by 3timeslazy** [comment]
Too many commits, so it's best to just watch ```Files changed```

File: pkg/stdlib/objects/merge.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+func mergeArray(arr *values.Array) *values.Object {
+	merged := values.NewObject()
+
+	var val core.Value
+	var k values.String
+	var obj *values.Object
+
+	for idx := values.NewInt(0); idx < arr.Length(); idx++ {
+		obj = arr.Get(idx).(*values.Object)
+		for _, key := range obj.Keys() {
```
Comment (ziflex) [inline pkg/stdlib/objects/merge.go]: Why don't you use ``Object.ForEach`` instead? You would eliminate 1 extra iteration at least. (by removing ``.Keys`` method)

File: pkg/stdlib/objects/merge.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+func mergeArray(arr *values.Array) *values.Object {
+	merged := values.NewObject()
+
+	var val core.Value
+	var k values.String
+	var obj *values.Object
+
+	for idx := values.NewInt(0); idx < arr.Length(); idx++ {
```
Comment (ziflex) [inline pkg/stdlib/objects/merge.go]: Wouldn't ``Array.ForEach`` be more easy to use and read?

File: pkg/stdlib/objects/merge.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		obj = arr.Get(idx).(*values.Object)
+		for _, key := range obj.Keys() {
+			k = values.NewString(key)
+			val, _ = obj.Get(k)
+			if values.IsCloneable(val) {
+				val = val.(core.Cloneable).Clone()
```
Comment (ziflex) [inline pkg/stdlib/objects/merge.go]: Do you think we really need to clone it? I was thinking of just copying.
Could you elaborate on that?

Comment (3timeslazy) [reply pkg/stdlib/objects/merge.go]: Variables in AQL are immutable once assigned. So, if we simply apply ```Copy```, then we will break this principle and we will be able to change the nested objects from the outside.

File: pkg/stdlib/objects/merge.go (hunk 1 above)
Comment (3timeslazy) [reply pkg/stdlib/objects/merge.go]: Always forget about this feature :) will fix it

File: pkg/stdlib/objects/merge.go (hunk 3 above)
Comment (3timeslazy) [reply pkg/stdlib/objects/merge.go]: So, do I continue to use the ```Clone``` function?

Comment (ziflex) [reply pkg/stdlib/objects/merge.go]: Yep.

**Pull request merged**